go/vt/sqlparser/token.go

```diff
@@ -63,7 +64,15 @@ func (p *Parser) NewStringTokenizer(sql string) *Tokenizer {
 // This function is used by go yacc.
 func (tkn *Tokenizer) Lex(lval *yySymType) int {
 	if tkn.SkipToEnd {
-		return tkn.skipStatement()
+		// We need to check the last token type to
+		// prevent us from skipping the next query in a multi
+		// parse mode. If we don't check the last token, we
+		// will skip the next query.
+		if tkn.lastTokenType == ';' {
+			tkn.SkipToEnd = false
+		} else {
+			return tkn.skipStatement()
+		}
```

Comment:
do we have test where we do partial and full parsing in multi statement?

Reply:
Yes

Reply:
looks like `TestSplitStatementToPieces` does it